ui/src/components/inputs/MonacoEditor.vue

```diff
@@ -147,21 +147,29 @@
                         modified: modifiedModel
                     });
                 } else {
-                    if (this.schemas !== undefined) {
-                        setDiagnosticsOptions({
-                            enableSchemaRequest: true,
-                            hover: true,
-                            completion: true,
-                            validate: true,
-                            format: true,
-                            schemas: this.schemas.map(r => {
-                                return {
-                                    uri: r,
-                                    fileMatch: ["*"]
-                                }
-                            })
-                        });
-                    }
+                    setDiagnosticsOptions({
+                        enableSchemaRequest: true,
+                        hover: true,
+                        completion: true,
+                        validate: true,
+                        format: true,
+                        schemas: [
+                            {
+                                fileMatch: ["flow.yaml"],
+                                uri: [`${apiRoot}plugins/schemas/flow`]
+                            },
+                            {
+                                fileMatch: ["task.yaml"],
+                                uri: [`${apiRoot}plugins/schemas/task`]
+                            },
+                            {
+                                fileMatch: ["template.yaml"],
+                                uri: [`${apiRoot}plugins/schemas/template`]
+                            }
+                        ]
+                    });
+
+                    options["model"] = monaco.editor.createModel(this.value, this.language, monaco.Uri.parse(`file:///${this.schemaType}.yaml`))
```

Comment:
add an uniqid on the model to allow multiple instance on the same page with the schemaType

Reply:
Monaco seems to (also?) look at the URI
So I know check if the model with the given URI already exists and return it instead of creating a new one